### ISSUE TITLE
FMIndex: ERROR: MethodError: Cannot `convert` an object of type String to an object of type Array{UInt8,1}

### ISSUE THREAD

**Issue by fpmenninger** [BioJulia/FMIndexes.jl#10]
> _This template is rather extensive. Fill out all that you can, if are a new contributor or you're unsure about any section, leave it unchanged and a reviewer will help you_ :smile:. _This template is simply a tool to help everyone remember the BioJulia guidelines, if you feel anything in this template is not relevant, simply delete it._

## Expected Behavior

```
julia> fmindex = FMIndex("abracadabra")
FMIndex{7,UInt8}:
     length: 11
  data size: 2.4 KiB
```
## Current Behavior

```
julia> fmindex = FMIndex("abracadabra")
ERROR: MethodError: Cannot `convert` an object of type String to an object of type Array{UInt8,1} Closest candidates are:
  convert(::Type{T<:Array}, ::AbstractArray) where T<:Array at array.jl:489
  convert(::Type{T<:AbstractArray}, ::T<:AbstractArray) where T<:AbstractArray at abstractarray.jl:14
  convert(::Type{T<:AbstractArray}, ::LinearAlgebra.Factorization) where T<:AbstractArray at C:\cygwin\home\Administrator\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.0\LinearAlgebra\src\factorization.jl:46
  ...
Stacktrace:
 [1] #FMIndex#2(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Type, ::String) at E:\ProgramFiles\Julia\.julia\packages\FMIndexes\yoEbF\src\FMIndexes.jl:80
 [2] FMIndex(::String) at E:\ProgramFiles\Julia\.julia\packages\FMIndexes\yoEbF\src\FMIndexes.jl:80
 [3] top-level scope at none:0
```
## Possible Solution / Implementation

```
79: function FMIndex(text::String; opts...)
80:    return FMIndex(convert(Vector{UInt8}, text), 128; opts...)
81: end
```
```
79: function FMIndex(text::String; opts...)
80:    return FMIndex([UInt8(symbol) for symbol in text]), 128; opts...)
81: end
```
Update 200322
Even better:
```
80:    return FMIndex(codeunits(text), 128; opts...)
```
## Context

Trying to run example code.

## Your Environment

- Package Version used: FMIndexes v1.0.0 #master (https://github.com/BioJulia/FMIndexes.jl)
- Julia Version used: julia-1.0.5 and julia-1.3.1
- Operating System and version (desktop or mobile): Windows 10 desktop
- Link to your project:

**Comment by tsgouros** [comment]
+1. 

Or 
```
79: function FMIndex(text::String; opts...)
80: return FMIndex(Vector{UInt8}(text), opts...)
81: end
```
You could also just change the example to read `fmindex=FMIndex(Vector{UInt8}("abracadabra"))` but this would be peculiar since the rest of the example works fine with ordinary strings.

**Comment by kescobo** [comment]
It's a bit weird, as this is the example that's in the README... one would think that would work. 

(PS - I edited your posts to fence the code to make it a bit easier to read, hope you don't mind)

**Comment by kescobo** [comment]
Oh, I see this code base is quite old - probably that's leftover from the days of implicit `convert` - would happily accept a PR to implement any of these changes.

**Comment by tsgouros** [comment]
Um, ok. I will submit a PR, but I've only been using Julia since Friday, so don't really know how to test the change I made. That is, I don't know how to tell Julia to use the FMIndexes I just edited instead of the one that I installed. When I use 'include' Julia complains about not having a package 'SuffixArrays' but the installed version of FMIndexes obviously uses that, so I am confused. Anyway please test it.

(And I agree with you that it's a bit odd for the first example in the README to be broken.)

**Comment by jakobnissen** [comment]
Fixed in #12, closing. You're welcome to re-open if you think more should be discussed.
